src/darker/tests/test_multiline_strings.py

```diff
@@ -0,0 +1,124 @@
+            "rf'single-quoted raw f-string'",
+            "b'single-quoted bytestring'",  # (line 5)
+            "rb'single-quoted raw bytestring'",
+            "'''triple-single-quoted one-line string'''",
+            "f'''triple-single-quoted one-line f-string'''",
+            "r'''triple-single-quoted one-line raw string'''",
+            "rf'''triple-single-quoted one-line raw f-string'''",  # (line 10)
+            "b'''triple-single-quoted one-line bytestring'''",
+            "rb'''triple-single-quoted one-line raw bytestring'''",
+            "'''triple-single-quoted",  # line 13
+            "   two-line string'''",
+            "f'''triple-single-quoted",  # line 15
+            "    two-line f-string'''",
+            "r'''triple-single-quoted",  # line 17
+            "    two-line raw string'''",
+            "rf'''triple-single-quoted",  # line 19
+            "     two-line raw f-string'''",
+            "b'''triple-single-quoted",  # line 21
+            "    two-line bytestring'''",
+            "rb'''triple-single-quoted",  # line 23
+            "     two-line raw bytestring'''",
+            '"double-quoted string"',  # (line 25)
+            'f"double-quoted f-string"',
+            'r"double-quoted raw string"',
+            'rf"double-quoted raw f-string"',
+            'b"double-quoted bytestring"',
+            'rb"double-quoted raw bytestring"',  # (line 30)
+            '"""triple-double-quoted one-line string"""',
+            'f"""triple-double-quoted one-line f-string"""',
+            'r"""triple-double-quoted one-line raw string"""',
+            'rf"""triple-double-quoted one-line raw f-string"""',
+            'b"""triple-double-quoted one-line bytestring"""',  # (line 35)
+            'rb"""triple-double-quoted one-line raw bytestring"""',
+            '"""triple-double-quoted',  # line 37
+            '   two-line string"""',
+            'f"""triple-double-quoted',  # line 39
+            '    two-line f-string"""',
+            'r"""triple-double-quoted',  # line 41
+            '    two-line raw string"""',
+            'rf"""triple-double-quoted',  # line 43
+            '     two-line raw f-string"""',
+            'b"""triple-double-quoted',  # line 45
+            '    two-line bytestring"""',
+            'rb"""triple-double-quoted',  # line 47
+            '     two-line raw bytestring"""',
+            '"""triple-',  # line 49
+            "   double-",
+            "   quoted",
+            "   six-",
+            "   line",
+            '   string"""',  # line 54
+        ]
+    )
+
+    result = multiline_strings.get_multiline_string_ranges(content)
+
+    assert result == [
+        # 1-based, end-exclusive
+        (13, 15),
+        (15, 17),
+        (17, 19),
+        (19, 21),
+        (21, 23),
+        (23, 25),
+        (37, 39),
+        (39, 41),
+        (41, 43),
+        (43, 45),
+        (45, 47),
+        (47, 49),
+        (49, 55),
+    ]
```

Comment:
Maybe, provide a message, to show the values/diff. Easier debugging.

Reply:
Here, too, for me Pytest already gives pretty verbose output if I change `(13, 15)` to `(12, 15)`:
```python
>       assert result == [
            # 1-based, end-exclusive
            (12, 15),
            (15, 17),
            (17, 19),
            (19, 21),
            (21, 23),
            (23, 25),
            (37, 39),
            (39, 41),
            (41, 43),
            (43, 45),
            (45, 47),
            (47, 49),
            (49, 55),
        ]
E       assert [(13, 15), (1...(23, 25), ...] == [(12, 15), (1...(23, 25), ...]
E         At index 0 diff: (13, 15) != (12, 15)
E         Use -v to get the full diff
```

And if I add the `-v` option, I get:
```python
E       AssertionError: assert [(13, 15), (1...(23, 25), ...] == [(12, 15), (1...(23, 25), ...]
E         At index 0 diff: (13, 15) != (12, 15)
E         Full diff:
E           [
E         -  (12, 15),
E         ?    ^
E         +  (13, 15),
E         ?    ^...
E         
E         ...Full output truncated (14 lines hidden), use '-vv' to show
```

...and with `-vv`:

```python
E       assert [(13, 15),\n (15, 17),\n (17, 19),\n (19, 21),\n (21, 23),\n (23, 25),\n (37, 39),\n (39, 41),\n (41, 43),\n (43, 45),\n (45, 47),\n (47, 49),\n (49, 55)] == [(12, 15),\n (15, 17),\n (17, 19),\n (19, 21),\n (21, 23),\n (23, 25),\n (37, 39),\n (39, 41),\n (41, 43),\n (43, 45),\n (45, 47),\n (47, 49),\n (49, 55)]
E         At index 0 diff: (13, 15) != (12, 15)
E         Full diff:
E           [
E         -  (12, 15),
E         ?    ^
E         +  (13, 15),
E         ?    ^
E            (15, 17),
E            (17, 19),
E            (19, 21),
E            (21, 23),
E            (23, 25),
E            (37, 39),
E            (39, 41),
E            (41, 43),
E            (43, 45),
E            (45, 47),
E            (47, 49),
E            (49, 55),
E           ]
```

Reply:
Check, I wasn't aware that assert in pytest gives a  good message